File: app/src/main/java/antoninovitale/dropcodechallenge/list/model/Attribute.java
package antoninovitale.dropcodechallenge.list.model;

/**
 * Created by antoninovitale on 31/08/2017.
 */
public enum Attribute {
    STRONG, BITTER, STRONG_BITTER, NONE;

    public static Attribute getAttribute(double abv, double ibu) {
        Attribute attribute;
        if (abv > 10 && ibu > 90) {
            attribute = Attribute.STRONG_BITTER;
        } else if (abv > 10) {
            attribute = Attribute.STRONG;
        } else if (ibu > 90) {
            attribute = Attribute.BITTER;
        } else {
            attribute = Attribute.NONE;
        }
        return attribute;
    }

    public static String getAttributeLabel(Attribute attribute, String attributeStrong, String
            attributeBitter) {
        String label = null;
        switch (attribute) {
            case STRONG:
                label = attributeStrong;
                break;
            case BITTER:
                label = attributeBitter;
                break;
            case STRONG_BITTER:
                label = String.format("%s & %s", attributeStrong, attributeBitter);
                break;
            default:
                break;
        }
        return label;
    }

}

File: app/src/main/java/com/ninovitale/punkapi/app/list/model/Attribute.kt
package com.ninovitale.punkapi.app.list.model

/**
 * Created by antoninovitale on 31/08/2017.
 */
enum class Attribute {
    STRONG, BITTER, STRONG_BITTER, NONE;

    companion object {
        fun getAttribute(abv: Double, ibu: Double): Attribute {
            return when {
                abv > 10 && ibu > 90 -> STRONG_BITTER
                abv > 10 -> STRONG
                ibu > 90 -> BITTER
                else -> NONE
            }
        }

        fun getAttributeLabel(
                attribute: Attribute?,
                attributeStrong: String?,
                attributeBitter: String?
        ): String? {
            var label: String? = null
            when (attribute) {
                STRONG -> label = attributeStrong
                BITTER -> label = attributeBitter
                STRONG_BITTER -> label = String.format("%s & %s", attributeStrong, attributeBitter)
                else -> {
                }
            }
            return label
        }
    }
}